pennylane/gradients/parameter_shift.py

```diff
@@ -582,6 +641,11 @@ def param_shift(
         :attr:`~.operation.Operation.parameter_frequencies`, and finally
         :meth:`~.operation.Operation.generator` via the default implementation of the frequencies.
 
+    .. warning::
+
+        Note that using parameter broadcasting via ``broadcast=True`` is not supported for tapes
+        with multiple return values or for evaluations with shot vectors.

```

Comment:
What does the user see if this is attempted? Is there a dedicated error that is raised?

Reply:
Now there is :) 
It should be noted that for _some_ combinations of return types and interface, `broadcast=True` actually would be supported. However, raising an error only exactly if it becomes problematic seems very difficult to me (unless we just catch any error raised by `param_shift`, but that seems dangerous to me).
This means that the now newly introduced `NotImplementedError` will be a bit too conservative, and disallow broadcasting in a few cases where it actually would work...

Reply:
Would it be safer though to forbid access until it works generally? 
Should we allow it to work on those few cases that it does, and accept some failures? 

I would tend to lean into preventing access to it until it is fully supported, but I can be swayed from this POV if the benefits are worth the lack of generality for now.

Reply:
I went for preventing the access. I second that clear-cut error messages and avoiding unexpected problems when changing a detail of a workflow (like switching the interface for some reason) is more important.